src/hermes/commands/deposit/invenio.py

```diff
@@ -17,648 +17,613 @@
-    }.items() if v is not None}
-
-    return deposition_metadata
-
-
-def _get_license_identifier(ctx: CodeMetaContext, license_api_url: str):
-    """Get Invenio license representation from CodeMeta.
-
-    The license to use is extracted from the ``license`` field in the
-    :class:`CodeMetaContext` and converted into an appropiate license identifier to be
-    passed to an Invenio instance.
-
-    A license according to CodeMeta may be a URL (text) or a CreativeWork. This function
-    only handles URLs. If a ``license`` field is present in the CodeMeta and it is not
-    of type :class:`str`, a :class:`RuntimeError` is raised.
-
-    Invenio instances take a license string which refers to a license identifier.
-    Typically, Invenio instances offer licenses from https://opendefinition.org and
-    https://spdx.org. However, it is possible to mint PIDs for custom licenses.
-
-    An API endpoint (usually ``/api/licenses``) can be used to check whether a given
-    license is supported by the Invenio instance. This function tries to retrieve the
-    license by the identifier at the end of the license URL path. If this identifier
-    does not exist on the Invenio instance, a :class:`RuntimeError` is raised. If no
-    license is given in the CodeMeta, ``None`` is returned.
-    """
-
-    license_url = ctx["codemeta"].get("license")
-
-    if license_url is None:
-        return None
-
-    if not isinstance(license_url, str):
-        raise RuntimeError(
-            "The given license in CodeMeta must be of type str. "
-            "Licenses of type 'CreativeWork' are not supported."
+        latest_record_id = self.invenio_ctx.get("latestRecord", {}).get("id")
+
+        # Get current deposit
+        response = self.client.get_deposit(latest_record_id)
+        if not response.ok:
+            raise RuntimeError(f"Failed to get current deposit {response.url!r}")
+
+        self.links.update(response.json()["links"])
+
+        # Create a new version using the newversion action
+        deposit_url = self.links["newversion"]
+        response = self.client.post(deposit_url)
+        if not response.ok:
+            raise RuntimeError(f"Could not create new version deposit {deposit_url!r}")
+
+        # Store link to latest draft to be used in :func:`update_metadata`.
+        old_deposit = response.json()
+        self.links.update(old_deposit["links"])
+
+    def update_metadata(self) -> None:
+        """Update the metadata of a draft."""
+
+        draft_url = self.links["latest_draft"]
+
+        deposition_metadata = self.invenio_ctx["depositionMetadata"]
+
+        response = self.client.put(
+            draft_url,
+            json={"metadata": deposition_metadata}
         )
 
-    parsed_url = urlparse(license_url)
-    url_path = parsed_url.path.rstrip("/")
-    license_id = url_path.split("/")[-1]
+        if not response.ok:
+            raise RuntimeError(f"Could not update metadata of draft {draft_url!r}")
```

Comment:
**chore:** Again, we need to know about the error message returned by the server.